src/ert/_c_wrappers/enkf/ensemble_config.py

```diff
@@ -91,165 +106,271 @@ def _load_refcase(self, refcase_file: Optional[str]) -> Optional[EclSum]:
+        field_list=None,
     ):
-        if config_content is not None and config_dict is not None:
-            raise ValueError(
-                "Attempting to create EnsembleConfig "
-                "object with multiple config objects"
-            )
-
-        self._grid_file: Optional[str] = None
-        self._refcase_file: Optional[str] = None
-        c_ptr = None
-
-        if config_dict is not None:
-            c_ptr = self._alloc_full(config_dict.get(ConfigKeys.GEN_KW_TAG_FORMAT))
-            grid_file_path = _get_abs_path(config_dict.get(ConfigKeys.GRID))
-            self._grid_file: Optional[str] = grid_file_path
-            refcase_file_path = _get_abs_path(config_dict.get(ConfigKeys.REFCASE))
-            self._refcase_file: Optional[str] = refcase_file_path
-            self.grid = self._load_grid(self._grid_file)
-            self.refcase = self._load_refcase(self._refcase_file)
-
-        if config_content is not None:
-            if config_content.hasKey(ConfigKeys.GRID):
-                self._grid_file = config_content.getValue(ConfigKeys.GRID)
-            if config_content.hasKey(ConfigKeys.REFCASE):
-                self._refcase_file = config_content.getValue(ConfigKeys.REFCASE)
-            self.grid = self._load_grid(self._grid_file)
-            self.refcase = self._load_refcase(self._refcase_file)
-
-            c_ptr = self._alloc(config_content, self.grid, self.refcase)
+        if gen_kw_list is None:
+            gen_kw_list = []
+        if gen_data_list is None:
+            gen_data_list = []
+        if surface_list is None:
+            surface_list = []
+        if summary_list is None:
+            summary_list = []
+        if schedule_file_list is None:
+            schedule_file_list = []
+        if field_list is None:
+            field_list = []
+
+        self._grid_file = grid_file
+        self._refcase_file = ref_case_file
+        self.grid: Optional[EclGrid] = self._load_grid(grid_file)
+        self.refcase: Optional[EclSum] = self._load_refcase(ref_case_file)
+        self._gen_kw_tag_format = tag_format
+        c_ptr = self._alloc_full(self._gen_kw_tag_format)
 
         if c_ptr is None:
             raise ValueError("Failed to construct EnsembleConfig instance")
         super().__init__(c_ptr)
 
-    @classmethod
-    def from_dict(cls, config_dict):
-        ens_config = cls(config_dict=config_dict)
-        gen_data_list = config_dict.get(ConfigKeys.GEN_DATA, [])
         for gene_data in gen_data_list:
-            gen_data_node = EnkfConfigNode.create_gen_data_full(
-                gene_data.get(ConfigKeys.NAME),
-                gene_data.get(ConfigKeys.RESULT_FILE),
-                gene_data.get(ConfigKeys.INPUT_FORMAT),
-                gene_data.get(ConfigKeys.REPORT_STEPS),
-            )
-            ens_config.addNode(gen_data_node)
+            node = self.gen_data_node(gene_data)
+            if node is not None:
+                self.addNode(node)
 
```

Comment:
This was the behaviour before the changes so I thought I would keep it as it was. If I throw an error now that will probably break some existing config files and I wanted to avoid that.

Reply:
sounds very fair indeed!